csrc/python_frontend/python_bindings.cpp

```diff
@@ -149,6 +149,46 @@ Tensor reshape_fn(
   return output;
 }
 
+template <class ShapeType, serde::RecordType RType>
+Tensor random_dist_op_fn(
+    FusionDefinition::Operators& self,
+    Scalar arg1,
+    Scalar arg2,
+    ShapeType generic_new_shape,
+    std::optional<Scalar> rng_seed,
+    std::optional<Scalar> rng_offset,
+    PrimDataType dtype) {
+  static_assert(
```

Comment:
nitpick. The assert here looks a bit strange.
RType is passed to RandomDistOpRecord and isn't used here. We can just rely on the assert in the template instead.

Reply:
RType is a template parameter and this should be a compile time assert.   I believe there is a better way to do this with C++ 20 but I am not inclined to use C++20 until we officially decide to use it.

Reply:
I put the `static_assert` in the `else` statement.